### ISSUE TITLE
DeepSpeed FusedLamb vs Apex FusedLamb

### ISSUE THREAD

**Issue by LiweiPeng** [microsoft/DeepSpeed#163]
Nvidia's apex also has a good implementation of [FusedLAMB](https://github.com/NVIDIA/apex/blob/master/apex/optimizers/fused_lamb.py). What's the advantage/disadvantage of using apex's FusedLAMB compared with DeepSpeed's lamb implementation?

**Comment by samyam** [comment]
Hi Liwei, There should be no disadvantage of using DeepSpeed's lamb implementation compared to Apex. On the other hand, there are certain cases where we have noticed an advantage with DeepSpeed's Lamb compared to Nvidia's apex. The DeepSpeed's lamb implementation allows the user the control the min and the max lamb coefficient value. We have found that tuning these parameters can lead to a better convergence time than using the lamb coefficients without any upper or lower bounds for very large batch sizes.
